inference-engine/src/multi_device/multi_device_async_infer_request.cpp

```diff
@@ -35,31 +38,42 @@ MultiDeviceAsyncInferRequest::MultiDeviceAsyncInferRequest(
         { /*TaskExecutor*/ std::make_shared<ImmediateExecutor>(), /*task*/ [this] {
-               // by default, no preferred device:
-               _multiDeviceExecutableNetwork->_thisPreferredDeviceName = "";
-               // if any input is remote (e.g. was set with SetBlob), let' use the corresponding device
-               for (const auto &it : _multiDeviceExecutableNetwork->GetInputsInfo()) {
-                   auto b = _inferRequest->GetBlob(it.first);
-                   auto r = b->as<RemoteBlob>();
-                   if (r) {
-                       const auto name = r->getDeviceName();
-                       const auto res = std::find_if(
-                               _multiDeviceExecutableNetwork->_devicePrioritiesInitial.cbegin(),
-                               _multiDeviceExecutableNetwork->_devicePrioritiesInitial.cend(),
-                               [&name](const MultiDevicePlugin::DeviceInformation& d){
-                                    return d.deviceName == name; });
-                       if (_multiDeviceExecutableNetwork->_devicePrioritiesInitial.cend() == res) {
-                           IE_THROW() << "None of the devices (for which current MULTI-device configuration was "
-                                                 "initialized) supports a remote blob created on the device named " << name;
+                // by default, no preferred device:
+                _multiDeviceExecutableNetwork->_thisPreferredDeviceName = "";
+                // if any input is remote (e.g. was set with SetBlob), let' use the corresponding device
+                for (const auto &it : _multiDeviceExecutableNetwork->GetInputsInfo()) {
+                    auto b = _inferRequest->GetBlob(it.first);
+                    auto r = b->as<RemoteBlob>();
+                    if (r) {
+                        const auto name = r->getDeviceName();
+                        const auto res = std::find_if(
+                                _multiDeviceExecutableNetwork->_devicePrioritiesInitial.cbegin(),
+                                _multiDeviceExecutableNetwork->_devicePrioritiesInitial.cend(),
+                                [&name](const MultiDevicePlugin::DeviceInformation& d) {
+                                    if (d.deviceName == name)
+                                        return true;
 
-                       } else {
+                                    // In case if remote blob always returns fully-specified device name (xPU.ID)
+                                    // we try to append default device ID to the name in DeviceInformation and compare again
+                                    DeviceIDParser parsed{d.deviceName};
+                                    if (parsed.getDeviceID().empty() && !d.defaultDeviceID.empty()) {
+                                        return (d.deviceName + "." + d.defaultDeviceID) == name;
```

Comment:
@myshevts should not we use such comparison in MultiExecNet.SetConfig?

Reply:
we discussed with Vladimir that this code should be move from scheduling to the ParseMetaDevices